File: src/KeyBoardRow.java
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class KeyBoardRow {
    private static final Map<Character, Integer> alphabetPosition = new HashMap<>();

    static {
        alphabetPosition.put('Q', 1);
        alphabetPosition.put('W', 1);
        alphabetPosition.put('E', 1);
        alphabetPosition.put('R', 1);
        alphabetPosition.put('T', 1);
        alphabetPosition.put('Y', 1);
        alphabetPosition.put('U', 1);
        alphabetPosition.put('I', 1);
        alphabetPosition.put('O', 1);
        alphabetPosition.put('P', 1);

        alphabetPosition.put('A', 2);
        alphabetPosition.put('S', 2);
        alphabetPosition.put('D', 2);
        alphabetPosition.put('F', 2);
        alphabetPosition.put('G', 2);
        alphabetPosition.put('H', 2);
        alphabetPosition.put('J', 2);
        alphabetPosition.put('K', 2);
        alphabetPosition.put('L', 2);

        alphabetPosition.put('Z', 3);
        alphabetPosition.put('X', 3);
        alphabetPosition.put('C', 3);
        alphabetPosition.put('V', 3);
        alphabetPosition.put('B', 3);
        alphabetPosition.put('N', 3);
        alphabetPosition.put('M', 3);
    }

    public static String[] findWords(String[] words) {
        List<String> result = new ArrayList<>();
        for (String word : words) {
            if (singleRowWord(word.toUpperCase())) {
                result.add(word);
            }
        }

        return arrayListToArray(result);
    }

    private static String[] arrayListToArray(List<String> list) {
        String[] result = new String[list.size()];
        for (int index = 0 ; index < result.length ; index++) {
            result[index] = list.get(index);
        }
        return result;
    }

    private static boolean singleRowWord(String word) {
        int row = alphabetPosition.get(word.charAt(0));
        for (int index = 1 ; index < word.length() ; index++) {
            if (row != alphabetPosition.get(word.charAt(index))) {
                return false;
            }
        }
        return true;
    }
}

File: python/keyboard_row.py
from typing import List


class Solution:
    def __init__(self):
        self.alphabetPosition = {
            'q': 1,
            'w': 1,
            'e': 1,
            'r': 1,
            't': 1,
            'y': 1,
            'u': 1,
            'i': 1,
            'o': 1,
            'p': 1,
            'a': 2,
            's': 2,
            'd': 2,
            'f': 2,
            'g': 2,
            'h': 2,
            'j': 2,
            'k': 2,
            'l': 2,
            'z': 3,
            'x': 3,
            'c': 3,
            'v': 3,
            'b': 3,
            'n': 3,
            'm': 3,
        }

    def findWords(self, words: List[str]) -> List[str]:
        result = []
        for word in words:
            if self.singleRowWord(word.lower()):
                result.append(word)
        return result

    def singleRowWord(self, word: str) -> bool:
        row = self.alphabetPosition[word[0]]
        for index in range(1, len(word)):
            if self.alphabetPosition[word[index]] != row:
                return False
        return True
